Lib/distutils/command/check.py

```diff
@@ -120,7 +120,7 @@ def check_restructuredtext(self):
 
     def _check_rst_data(self, data):
         """Returns warnings when the provided data doesn't compile."""
-        source_path = StringIO()
+        source_path = self.distribution.script_name or 'long_description'
```

Comment:
Do you think it would be better to have a name looking like `<something>`, to hint that this is not a real filename?

(I forget which stdlib module uses `<stdin>` but I like the convetion)

Reply:
so the data is always retrieved via `self.distribution.get_long_description()`, so some variant of `long_description` seems to be most informative to me.

maybe something like this?

```py
source_path = '<long_description>'
if self.distribution.script_name:
    source_path = '{} {}'.format(self.distribution.script_name, source_path)
```

Reply:
That depends if source_path has to be a path-like string, or arbitrary text.

I see your point that a warning or error message saying only «setup.py: bad include bla bla» may not be enough to let the user understand that the issue is reST in long_description, especially if the text is read from a README or other file.

If source_path is for human display only and doesn’t have to look like a file path, then +1 to your idea, maybe with parens rather than brackets: `setup.py (long_description)` / fallback `distutils (long_description)`

Reply:
at multiple points the attribute docs say something like “Path to **or description** of the input source being processed”, which makes me assume that free-form text is OK.